src/software/pipeline/main_cameraInit.cpp

```diff
@@ -396,37 +399,78 @@ int main(int argc, char **argv)
     }
 
     // get view intrinsic sensor width
-    if(hasCameraMetadata)
     {
-      aliceVision::sensorDB::Datasheet datasheet;
-      if(getInfo(make, model, sensorDatabase, datasheet))
+      // try to find in the sensor database
+      if(hasCameraMetadata)
       {
-        sensorWidth = datasheet._sensorSize; // sensor is in the database
+        aliceVision::sensorDB::Datasheet datasheet;
+        if(getInfo(make, model, sensorDatabase, datasheet))
+        {
+          // sensor is in the database
+          ALICEVISION_LOG_DEBUG("Sensor width found in database: " << std::endl
+                                << "\t- brand: " << make << std::endl
+                                << "\t- model: " << model << std::endl
+                                << "\t- sensor width: " << datasheet._sensorSize << " mm");
+
+          if(datasheet._model != model) // the camera model in database is slightly different
+            unsureSensors.emplace(std::make_pair(make, model), std::make_pair(view.getImagePath(), datasheet)); // will throw a warning message
+
+          sensorWidth = datasheet._sensorSize;
+        }
       }
-      else
+
+      // try to find / compute with 'FocalLengthIn35mmFilm' metadata
+      if(sensorWidth == -1.0 && hasFocalIn35mmMetadata)
       {
-        #pragma omp critical
-        unknownSensors.emplace(std::make_pair(make, model), view.getImagePath()); // will throw an error message
-        if(!allowIncompleteOutput)
-          continue;
+        const double focalIn35mm = std::stod(view.getMetadata("Exif:FocalLengthIn35mmFilm"));
+        const double imageRatio = std::min(view.getWidth(), view.getHeight()) / std::max(view.getWidth(), view.getHeight());
+
+        if(focalLength > 0.0)
+        {
+          const double sensorDiag = (focalLength * 43.3) / focalIn35mm; // 43.3 is the diagonal of 35mm film
+          sensorWidth = std::sqrt((1.0/(1.0 + imageRatio * imageRatio)) * sensorDiag * sensorDiag);
```

Comment:
Remove the useless `1.0/`:
`sensorWidth = std::sqrt((sensorDiag * sensorDiag) / (1.0 + imageRatio * imageRatio));`

Reply:
or even 
```
sensorWidth = sensorDiag * std::sqrt(1.0 / (1.0 + imageRatio * imageRatio));
```